sdkproject/Assets/Mapbox/Unity/MeshGeneration/LayerVisualizers/VectorLayerVisualizer.cs

```diff
@@ -215,91 +322,123 @@ private IEnumerator ProcessLayer(VectorTileLayer layer, UnityTile tile, Action c
 				yield break;
 			}
 
-			//testing each feature with filters
-			var fc = layer.FeatureCount();
+			Debug.Log("Tile Id -> " + tile.name + "Layer Name -> " + layer.Name + " features -> " + layer.FeatureCount());
+			if (!_vectorFeaturesPerTile.ContainsKey(tile))
+			{
+				_vectorFeaturesPerTile.Add(tile, new VectorLayerVisualizerProperties
+				{
+					vectorTileLayer = layer,
+				});
+			}
+
 			//Get all filters in the array.
-			var filters = _layerProperties.filterOptions.filters.Select(m => m.GetFilterComparer()).ToArray();
+			_vectorFeaturesPerTile[tile].layerFeatureFilters = _layerProperties.filterOptions.filters.Select(m => m.GetFilterComparer()).ToArray();
 
 			// Pass them to the combiner
-			Filters.ILayerFeatureFilterComparer combiner = new Filters.LayerFilterComparer();
+			_vectorFeaturesPerTile[tile].layerFeatureFilterCombiner = new Filters.LayerFilterComparer();
 			switch (_layerProperties.filterOptions.combinerType)
 			{
 				case Filters.LayerFilterCombinerOperationType.Any:
-					combiner = Filters.LayerFilterComparer.AnyOf(filters);
+					_vectorFeaturesPerTile[tile].layerFeatureFilterCombiner = Filters.LayerFilterComparer.AnyOf(_vectorFeaturesPerTile[tile].layerFeatureFilters);
```

Comment:
What's the advantage of removing the variable `combiner` and writing `_vectorFeaturesPerTile[tile].layerFeatureFilterCombiner` instead? Same question for `filters`.

Reply:
Caching the variables, so we don't have to compute the values multiple times + avoid passing 5 variables to every function.